test/cri-containerd/runpodsandbox_test.go

```diff
@@ -71,6 +77,63 @@ func Test_RunPodSandbox_LCOW(t *testing.T) {
 	runPodSandboxTest(t, request)
 }
 
+func Test_RunPodSandbox_Events_LCOW(t *testing.T) {
+	client := newTestRuntimeClient(t)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	podctx, podcancel := context.WithCancel(context.Background())
+	defer podcancel()
+
+	pullRequiredLcowImages(t, []string{imageLcowK8sPause})
+
+	request := &runtime.RunPodSandboxRequest{
+		Config: &runtime.PodSandboxConfig{
+			Metadata: &runtime.PodSandboxMetadata{
+				Name:      t.Name(),
+				Uid:       "0",
+				Namespace: testNamespace,
+			},
+		},
+		RuntimeHandler: lcowRuntimeHandler,
+	}
+
+	topicNames, filters := getTargetRunTopics()
+	targetNamespace := "k8s.io"
+
+	eventService := newTestEventService(t)
+	stream, errs := eventService.Subscribe(ctx, filters...)
```

Comment:
As discussed, we are slightly unsure where the event buffering is coming from, so let's avoid it altogether...

I would suggest moving the code that reads each event in turn and validates it's what we expect (the "state machine") into a goroutine. This goroutine would share an error channel with the main func. If an error is encountered in the goroutine, instead of calling t.Fatal, it would instead write an error to the channel. When the state machine completes successfully, close the channel. Outside of the goroutine, read from the goroutine, and call t.Fatal if you got an error, or exit successfully if you got nil.